### PR TITLE
[capi/util] Change framework auto-detection

### PR DESCRIPTION
**Pull request by gichan-jang**
Integrate duplicated code for framework auto detection.

Signed-off-by: gichan-jang <gichan2.jang@samsung.com>

**Self evaluation:**
1. Build test: [ *]Passed [ ]Failed []Skipped
2. Run test: [ *]Passed [ ]Failed []Skipped

### CODE REVIEW

File: api/capi/src/nnstreamer-capi-util.c [hunk 1]
```diff
@@ -933,46 +935,35 @@ ml_validate_model_file (char **model, unsigned int num_models,
...
-          ml_logi ("The given model [%s] is supposed a tensorflow-lite model.",
-              model[0]);
-          *nnfw = ML_NNFW_TYPE_TENSORFLOW_LITE;
-        }
-      } else if ((g_str_has_suffix (path_down, ".so") &&
-              g_str_has_suffix (path_down_2, ".nb")) ||
-          (g_str_has_suffix (path_down, ".nb") &&
-              g_str_has_suffix (path_down_2, ".so"))) {
+      detected_fw =
+          gst_tensor_filter_framework_auto_detection ((const gchar **) model,
```
Comment (wooksong) [inline api/capi/src/nnstreamer-capi-util.c:939]: It looks that gst_tensor_filter_framework_auto_detection returns a g_strdup()ed -gchar. 

Q. Is it freed after all?

Comment (gichan-jang) [reply api/capi/src/nnstreamer-capi-util.c:939]: Right. detected_fw should be released. I will update it.

File: api/capi/src/nnstreamer-capi-util.c [hunk 2]
```diff
@@ -933,46 +935,35 @@ ml_validate_model_file (char **model, unsigned int num_models,
...
+            model[0]);
+        status = ML_ERROR_INVALID_PARAMETER;
+      } else if (g_ascii_strcasecmp (detected_fw, "nnfw") == 0) {
+        ml_logi ("The given model [%s] is supposed a nnfw model.", model[0]);
+        *nnfw = ML_NNFW_TYPE_NNFW;
+      } else if (g_ascii_strcasecmp (detected_fw, "tensorflow-lite") == 0) {
+        ml_logi ("The given model [%s] is supposed a tensorflow-lite model.",
+            model[0]);
+        *nnfw = ML_NNFW_TYPE_TENSORFLOW_LITE;
+      } else if (g_ascii_strcasecmp (detected_fw, "vivante") == 0) {
```
Comment (jaeyun-jung) [inline api/capi/src/nnstreamer-capi-util.c:953]: I think you missed one line,
*nnfw = ML_NNFW_TYPE_VIVANTE;

File: api/capi/src/nnstreamer-capi-util.c [hunk 3]
```diff
@@ -27,6 +27,7 @@
 #include "nnstreamer_plugin_api.h"
 #include "nnstreamer_plugin_api_filter.h"
 #include "nnstreamer_conf.h"
+#include <nnstreamer/tensor_filter/tensor_filter_common.h>
```
Comment (myungjoo) [inline api/capi/src/nnstreamer-capi-util.c:30]: ```#include <tensor_filter/tensor_filter_common.h>``` should work.

File: api/capi/src/nnstreamer-capi-util.c (hunk 2 above)
Comment (gichan-jang) [reply api/capi/src/nnstreamer-capi-util.c:953]: Oh.. Thank you. Updated

File: api/capi/src/nnstreamer-capi-util.c (hunk 3 above)
Comment (gichan-jang) [reply api/capi/src/nnstreamer-capi-util.c]: Thank you. Updated!

**Comment by gichan-jang** [comment]
> LGTM.
> 
> Add test cases that use different nnstreamer.ini file contents of [filter]/"framework_priority_tflite" in another PR. In `tests/common/unittest_common.c`, there are other test cases doing similar things. You can refer them.

Understand. I will add test case with different priorities of the frameworks.

**Pull request merged**